src/request-handler.ts

```diff
@@ -165,6 +164,35 @@ export function createRequestHandler(
     }
   };
 
+  /**
+   * Sets the Content-Type header if needed based on the method and body.
+   *
+   * @param {HeadersInit} headers - The headers object where Content-Type will be set.
+   * @param {string} method - The HTTP method (e.g., GET, POST, PUT, DELETE).
+   * @param {any} [body] - Optional request body to determine if Content-Type is needed.
+   */
+  const setContentTypeIfNeeded = (
+    headers: HeadersInit,
+    method: string,
+    body?: any,
+  ) => {
+    // For PUT and DELETE methods, do not set Content-Type if no body is provided.
+    if (['PUT', 'DELETE'].includes(method) && !body) {
```

Comment:
Small performance improvement: we could check !body before calling the includes(). Might be faster for all requests that have body specified.

Reply:
That’s a good point!

Reply:
```suggestion
    if (!body && ['PUT', 'DELETE'].includes(method)) {
```
I mean I was suggesting this one, not sure if you forgot 😄 I think we can merge and release after this change

Reply:
@saewoohan I think it was all good, I was about to merge it in fact 😄 Can you reopen please?